File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int low=0,ans=0;
        unordered_map<char,int>m;
        for(int high=0;high<s.size();high++){
            m[s[high]]++;
            while(m[s[high]]>1){
                if(m[s[low]]==1)m.erase(s[low]);
                else m[s[low]]--;
                low++;
            }
            ans = max(ans,high-low+1);
        }
        return ans;
        
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {

    const umap = {}
    let cnt =0;
    let ans =1;
    let left =0;
    if(s.length ===0)return 0;
    umap[s[0]]=1;
    for(let right=1;right<s.length;right++){
        while(umap[s[right]]){
            delete umap[s[left]];
            left++
        }
        umap[s[right]] = 1;
        ans = Math.max(ans,right-left+1);
    }
    return ans;
};